packages/flutter_test/lib/src/controller.dart

```diff
@@ -234,6 +235,32 @@ abstract class WidgetController {
   /// A reference to the current instance of the binding.
   final WidgetsBinding binding;
 
+  /// The [TestPlatformDispatcher] that is being used in this test.
+  ///
+  /// This will be injected into the framework such that calls to
+  /// [WidgetsBinding.platformDispatcher] will use this. This allows
+  /// users to change platform specific properties for testing.
+  ///
+  /// See also:
+  ///
+  ///   * [TestFlutterView] which allows changing view specific properties
+  ///     for testing
+  ///   * [implicitView] and [viewOf] which are used to find
+  ///     [TestFlutterView]s from the widget tree
+  TestPlatformDispatcher get platformDispatcher => binding.platformDispatcher as TestPlatformDispatcher;
+
+  /// The implicit [TestFlutterView] provided when the platform does not support
+  /// creating windows.
+  ///
+  /// See also:
+  ///
+  ///   * [PlatformDispatcher.implicitView], which is the analog to this
+  ///     property in non-test environments.
+  ///   * [viewOf], which can find a TestFlutterView related to a given finder.
+  ///     This is how to modify view properties for testing when dealing with
+  ///     multiple views.
+  TestFlutterView get implicitView => platformDispatcher.implicitView!;
```

Comment:
Is that going to require new methods for each part of the API that _uses_ pump as well? That seems _very_ heavy handed if that's what we're planning...

Reply:
> Is that going to require new methods for each part of the API that uses pump as well?

I am not sure I understand this question. What do you mean?

My philosophy has been that flutter_test is a unit testing framework you use to test the functionality of widgets. You wanna test that your account picker widget can switch between accounts or that your form widget properly reports errors when someone enters letters into the phone number box. For all these tests you don't really care about views or windows - and you really don't want to deal with the complexity and boilerplate of creating and managing them for your test. You just want to render a widget under test somewhere and interact with it. For that, flutter_test provides you (behind the scenes) with a default FlutterView into which `tester.pumpWidget` renders your widget. Where flutter_test obtains that FlutterView from, shouldn't be your concern as a test author. It may be that flutter_test will create a window to render your widget under test into. It may be that the embedder just hands it a FlutterView to use (e.g. from PlatformDispatcher.implicitView). Doesn't matter. As a test author you just call `pumpWidget` as you always have to render the widget you want to test _somewhere_. To support this, there will always be a default view provided by flutter_test no matter if you're running on a platform that supports windows or not.

The widget passed to `pumpWidget` and rendered into that view may create additional auxiliary views (e.g. to show a tooltip). So, `pumpWidget` will support most relevant multi window testing use cases as well.

There is one thing `pumpWidget` cannot do in this setup: If you need two (or more) top-level views for your testing scenario `pumpWidget` will not do because it inherently assumes that the provided FlutterView is the single root of the view hierarchy. If this isn't what you want (and I expect these cases to be rather rare given how people usually write widget tests) you need to manually manage your view/window widgets and instead of passing them to `pumpWidget` pass them to something that doesn't wrap them in the default view (with `wrapWithDefaultView`) because you explicitly don't want the default view behavior. For this scenario (which again I think will be rare) we need an additional method `pumpWeNeedToFigureOutAGoodName` which unlike `pumpWidget` does not call `wrapWithDefaultView` and instead expects that the provided widget tree already contains the relevant view/window widgets. For this particular scenario, you'd call this method instead of `pumpWidget` and everything else stays the same.

Long story short: This is why I think that the property that holds this default FlutterView used by `pumpWidget` should be documented without mentioning window support. As I tried to explain above, the behavior of this property doesn't change whether windows are supported or not. What may change is the implementation of the property (i.e. where flutter_test obtains the FlutterView from that backs the property), but that is of no concern to the user of the property and shouldn't be the center of the doc comment.

Last, but not least: The name. We named `PlatformDispatcher.implicitView` that to kinda distract a little bit from it. We don't want people to find and use it. It's a low-level thing used to bootstrap the framework and kinda paper over the fact that we have platforms where the embedder will just provide us with a view and others where we will have to create a window first. This property is different. It's a proud property that we want people to use if their test scenario requires it. It's also always there (unlike `PlatformDispatcher.implicitView`). So, I think it deserves a straight-forward name like `view` or `testView`. Also to ensure that `PlatformDispatcher.implicitView` can continue to hide in the corner without drawing any attention.

Reply:
Ahh, so it's only `pumpWidget` that needs a multi-view alternative, not the lower level `pump` method? My concern was that we'd have to change _everything_ that uses pump for multi-view as well. If we can do `pumpWeNeedToFigureOutAGoodName` and then continue using the other methods that use `pump` afterwards, then I'm not as worried about the API.

So, would it make sense to change the implementation from `platformDispatcher.implicitView` to `platformDispatcher.views.single` then? It should be logically equivalent for now, but won't drive people investigating it towards `implicitView` and may be more flexible as we support other situations.

I'm still missing something about where the test framework is _getting_ it's views, because we're just wrapping them here. Does the engine provide a fake default `FlutterView` implementation during testing? Are there docs about the bootstrapping process? I've got a pretty good idea of the final structure (I think) but how the startup actually happens seems to be all over the place when I try to trace it...

Reply:
> Ahh, so it's only pumpWidget that needs a multi-view alternative, not the lower level pump method?

Correct.

> So, would it make sense to change the implementation from platformDispatcher.implicitView to platformDispatcher.views.single then?

I would say that using `PlatformDispatcher.implicitView` is still the right thing because as of right now (without true multi window support in place) that is the view the engine specifically provided for this purpose. It should just be an implementation detail and not be advertised to the users of the testing API as we may switch out that detail later (see below). While this is of course functionally identically to `PlatformDispatcher.views.single`, semantically `PlatformDispatcher.implicitView` makes a little more sense to me.

> I'm still missing something about where the test framework is getting it's views

In the future the testing framework in its bootstrapping process (before any test runs) will check whether the engine provided an implicit view to use (`PlatformDispatcher.implicitView`). If that's available, great, it will just use this (basically the same thing you implemented today). If it is not available, the test framework will talk to the yet-to-be-defined windowing API of the engine and ask for a window to be created. The View backing that window will then be used for the tests instead of the `PlatformDispatcher.implicitView`.

Reply:
I did end up going with `platformDispatcher.views.single`. I also added in an assertion that'd provide better messaging in the case where someone used `view` where they should've used `viewOf` instead. With that assertion, `platformDispatcher.views.single` reads better.

I also think that if we're trying to distance this method from `implicitView`, we may as well make it a convenience property for "modifying a single view" regardless of the method the test used to create that view.